app/models/CategoryRecord.scala

```diff
@@ -51,16 +50,25 @@ object CategoryRecord {
       CategoryRecord(
         eori = eori,
         recordId = recordId,
-        comcode = getLongerCommodityCode(answers, recordId),
         category = categoryDetails.category,
         categoryAssessmentsWithExemptions = categoryDetails.categoryAssessmentsWithExemptions,
         supplementaryUnit = supplementaryUnit,
         measurementUnit = measurementUnit
       )
     )
 
-  private def getLongerCommodityCode(answers: UserAnswers, recordId: String): Option[String] =
-    answers.get(LongerCommodityQuery(recordId)).map(_.commodityCode)
+  def buildForNiphls(eori: String, recordId: String, traderProfile: TraderProfile): CategoryRecord = {
+    val category = if (traderProfile.niphlNumber.isDefined) {
+      Category2
+    } else {
+      Category1
+    }
+
+    val categoryAsNumber = if (category == Category1) CATEGORY_1 else CATEGORY_2
+
+    CategoryRecord(eori, recordId, categoryAsNumber, 1)
```

Comment:
do we want to return the categoryAsNumber here instead of 1?

Reply:
We actually want the number of assessments they answered. Replaced it with that